### PR TITLE
ported disklayer account/storage iterator

### PR DESCRIPTION
**Pull request by jeongkyun-oh**
## Proposed changes

- ported disklayer structure
- ported account/storage iterator with tests
- introduced snapshot database with read/write/delete snapshot methods

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by aidan-kwon — APPROVED** [review]
LGTM

File: common/bytes_test.go [hunk 1]
```diff
@@ -103,3 +103,41 @@ func TestNoPrefixShortHexOddLength(t *testing.T) {
...
 	}
 }
+
+func TestTrimLeftZeroes(t *testing.T) {
+	tests := []struct {
+		arr []byte
+		exp []byte
+	}{
+		{FromHex("0x00ffff00ff0000"), FromHex("0xffff00ff0000")},
+		{FromHex("0x00000000000000"), []byte{}},
```
Comment (yoomee1313) [inline common/bytes_test.go:113]: What about adding more test cases, such as `0x0ffff00ff00000`?

Comment (yoomee1313) [inline common/bytes_test.go:113]: What about adding more test cases, such as `0x0ffff00ff00000`?

File: common/bytes_test.go [hunk 2]
```diff
@@ -103,3 +103,41 @@ func TestNoPrefixShortHexOddLength(t *testing.T) {
...
+	}
+}
+
+func TestTrimRightZeroes(t *testing.T) {
+	tests := []struct {
+		arr []byte
+		exp []byte
+	}{
+		{FromHex("0x00ffff00ff0000"), FromHex("0x00ffff00ff")},
+		{FromHex("0x00000000000000"), []byte{}},
```
Comment (yoomee1313) [inline common/bytes_test.go:132]: What about adding more test cases, such as `0x0ffff00ff00000`?

File: snapshot/iterator_test.go [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+import (
+	"bytes"
+	"io/ioutil"
+	"math/rand"
+	"os"
+	"testing"
+
+	"github.com/klaytn/klaytn/storage/database"
+
```
Comment (yoomee1313) [inline snapshot/iterator_test.go:31]:
```suggestion
```

**Review by yoomee1313 — COMMENTED** [review]
@jeongkyun-oh

File: common/bytes_test.go (hunk 1 above)
Comment (jeongkyun-oh) [reply common/bytes_test.go:114]: @yoomee1313 I added another test case. PTAL.

File: common/bytes_test.go (hunk 2 above)
Comment (jeongkyun-oh) [reply common/bytes_test.go:134]: @yoomee1313 I added another test case. PTAL.

File: snapshot/iterator_test.go (hunk 3 above)
Comment (jeongkyun-oh) [reply snapshot/iterator_test.go]: @yoomee1313 updated.

**Review by yoomee1313 — APPROVED** [review]

**Comment by jeongkyun-oh** [comment]
@hqjang-pepper Please let me know if you have another opinion.

**Pull request merged**